opentracing_instrumentation/client_hooks/_dbapi2.py

```diff
@@ -119,13 +161,15 @@ def __call__(self, *args, **kwargs):
 class ConnectionWrapper(wrapt.ObjectProxy):
     __slots__ = ('_module_name', '_connect_params')
 
+    cursor_cls = CursorWrapper
+
     def __init__(self, connection, module_name, connect_params):
         super(ConnectionWrapper, self).__init__(wrapped=connection)
         self._module_name = module_name
         self._connect_params = connect_params
 
     def cursor(self, *args, **kwargs):
-        return CursorWrapper(
+        return self.cursor_cls(
```

Comment:
`cursor_cls` is defined as class field, but you're referencing it as member field via `self.`

Reply:
Hi @yurishkuro,

You have to use `self` to access `cursor_cls` (that might be overridden) since `cursor` is not a `classmethod` and you have no `cls` here.
Surely, you can use `self.__class__.cursor_cls`, but it's completely redundant.

So it's a normal approach, there is nothing bad here actually.

Reply:
Yes, it is a common pattern in python to access a class field through instance of the class.

Reply:
A common pattern doesn't mean it's a good pattern.

Why not do it cleanly as a member variable initialized with value passed to the constructor?

Reply:
You are right, I did it with additional argument in `ConnectionFactory` constructor.